control-plane/api-gateway/binding/route_binding.go

```diff
@@ -227,6 +241,11 @@ func (r *Binder) dropConsulRouteParent(snapshot *Snapshot, object client.Object,
 }
 
 func (r *Binder) mutateRouteWithBindingResults(snapshot *Snapshot, object client.Object, gatewayConsulKey api.ResourceReference, resources *common.ResourceMap, results parentBindResults) {
+	if results.boundSections().Cardinality() == 0 {
+		r.dropConsulRouteParent(snapshot, object, r.nonNormalizedConsulKey, r.config.Resources)
+		return
+	}
```

Comment:
this allows us to clean up the parents of a route when it has bound to nothing and also has no actual references to the gateway anymore.